proto/regen/ecocredit/v1/tx.proto

```diff
@@ -57,6 +58,10 @@ message MsgCreateClass {
 
   // credit_type_abbrev describes the abbreviation of a credit type (e.g. "C", "BIO").
   string credit_type_abbrev = 4;
+
+  // fee specifies the fee to pay for the creation of the credit class.
+  // acceptable fees for creating a credit class can be found in the governance parameters for the ecocredit module.
+  Fee fee = 5;
```

Comment:
1. why someone would pay a different fee than the required one? Maybe we can use only `string denom` here and charge the required fee. 
2. If we really need to use both denom and amount, then let's use `Coin` type and remove Fee message. We can make it required (it's very small, so we in fact optimize the performance) to avoid null checks.

Reply:
1. i believe the intent was to ensure users aren't being charged fees that they are not explicitly setting. this was a concern i had also considered, so might be nice to get more opinions in here 

2. i changed it to coin, but im not sure about the nullable thing. pulsar doesn't support that (not sure if thats planned either?). it might be best to leave out the option knowing we we will switch fully to pulsar eventually

Reply:
since we are automatically adjusting, then maybe let's update the docs if we want to keep this attribute.